illumos-utils/src/zone.rs

```diff
@@ -417,46 +365,121 @@ impl Zones {
         Ok(())
     }
 
-    /// Returns all zones that may be managed by the Sled Agent.
-    ///
-    /// These zones must have names starting with [`ZONE_PREFIX`].
-    pub async fn get() -> Result<Vec<zone::Zone>, AdmError> {
-        Ok(zone::Adm::list()
-            .await
-            .map_err(|err| AdmError {
-                op: Operation::List,
-                zone: "<all>".to_string(),
-                err: err.into(),
-            })?
-            .into_iter()
-            .filter(|z| z.name().starts_with(ZONE_PREFIX))
-            .collect())
-    }
-
-    /// Finds a zone with a specified name.
-    ///
-    /// Can only return zones that start with [`ZONE_PREFIX`], as they
-    /// are managed by the Sled Agent.
-    pub async fn find(name: &str) -> Result<Option<zone::Zone>, AdmError> {
-        Ok(Self::get().await?.into_iter().find(|zone| zone.name() == name))
-    }
-
     /// Return the ID for a _running_ zone with the specified name.
-    //
-    // NOTE: This mostly exists for testing purposes. It's simple enough to call
-    // `Zones::find()` and then use the `zone::Zone::id()` method on that
-    // object. But that can't easily be done, because we need to supply
-    // `mockall` with a value to return, and `zone::Zone` objects can't be
-    // constructed since they have private fields.
-    pub async fn id(name: &str) -> Result<Option<i32>, AdmError> {
+    async fn id(
+        &self,
+        name: &str,
+    ) -> Result<Option<i32>, crate::zone::AdmError> {
         // Safety: illumos defines `zoneid_t` as a typedef for an integer, i.e.,
         // an `i32`, so this unwrap should always be safe.
-        match Self::find(name).await?.map(|zn| zn.id()) {
+        match self.find(name).await?.map(|zn| zn.id()) {
             Some(Some(id)) => Ok(Some(id.try_into().unwrap())),
             Some(None) | None => Ok(None),
         }
     }
 
+    async fn wait_for_service(
```

Comment:
Strictly speaking, this isn't a "zone" command, but it's tightly coupled with the zone boot process